### PR TITLE
Add support of GSI 5

### LINKED ISSUE #3609: Add support for GoogleSignIn 5
GoogleSignIn released a major update today and it causes build failures like 

❌  /Users/travis/build/firebase/firebase-ios-sdk/Example/Auth/Sample/GoogleAuthProvider.m:38:62: cannot find protocol declaration for 'GIDSignInUIDelegate'; did you mean 'GIDSignInDelegate'?
@interface GoogleAuthDelegate : NSObject <GIDSignInDelegate, GIDSignInUIDelegate, OpenURLDelegate>
                                                               ^
❌  /Users/travis/build/firebase/firebase-ios-sdk/Example/Auth/Sample/GoogleAuthProvider.m:85:38: no visible @interface for 'GIDSignIn' declares the selector 'handleURL:sourceApplication:annotation:'
  return [[GIDSignIn sharedInstance] handleURL:url
          ^
❌  /Users/travis/build/firebase/firebase-ios-sdk/Example/Auth/Sample/GoogleAuthProvider.m:117:10: property 'uiDelegate' not found on object of type 'GIDSignIn *'; did you mean 'delegate'?
  signIn.uiDelegate = delegate;
          ~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~~~~~~~~~~~~

See full log at https://travis-ci.org/firebase/firebase-ios-sdk/jobs/572459687

### PR DESCRIPTION
**Pull request by renkelvin**
Fix #3609

### CODE REVIEW

**Comment by paulb777** [comment]
Is it possible to support both GSI 4 and 5? Or do we need to choose one?

**Comment by paulb777** [comment]
Also, update the two Podfile's changed in #3608 accordingly

**Comment by renkelvin** [comment]
Our core SDK doesn't depend on GSI, so the version is fine.
The break is caused by the sample App. I think it's ok to use GSI 5.

File: Example/Podfile [hunk 1]
```diff
@@ -133,7 +133,7 @@ target 'Auth_Sample' do
   pod 'FirebaseCoreDiagnostics', :path => '../'
   pod 'FirebaseCoreDiagnosticsInterop', :path => '../'
   pod 'FBSDKLoginKit'
-  pod 'GoogleSignIn', '~> 4.4'
+  pod 'GoogleSignIn'
```
Comment (paulb777) [inline Example/Podfile]: Should it be , '~> 5.0'

Comment (renkelvin) [reply Example/Podfile]: Then we won't always get the latest version right?

Comment (paulb777) [reply Example/Podfile]: Right - I suppose this is better so that we catch it when 6.0 happens :)

**Pull request merged**